bohra/modules/collation/bin/collate_stats.py

```diff
@@ -30,41 +26,28 @@ def get_vals_kit(f):
 
     return int(reads.strip()), float(minval.strip()), float(avgval.strip()), float(maxval.strip())
 
-def get_length(ref):
+def get_length(genome_size):
 
-    p = subprocess.run(f"any2fasta {ref} | seqkit stats -a -T | cut -f5 | sed 1d", shell = True, capture_output = True, encoding = "utf-8")
-    length = int(p.stdout.strip())
-    print(p)
-    return length
-
-def get_dpth(ref,bases):
-    length = get_length(ref)
+    with open(genome_size, 'r') as f:
+        x = f.read().strip().split('\n')[0].strip()
+        return int(x)
```

Comment:
You are correct @abcdtree .. i want it to fall over though.. because if the genome size is not an int there are bigger problems.. BUT i have added in a new commit below to make it fall over more elegantly .. rather than just blow up.